promising/promise.py

```
@@ -330,30 +338,31 @@ async def _fulfill(self) -> None:
-            else:
+            if exception is None:
                 self.set_result(result)
+            else:
+                self.set_exception(exception)
 
     def _ensure_task_scheduled(self) -> None:
         if self._task is None and not self.done():
             self._task = self._ctx_loop.create_task(self._fulfill(), name=str(self) + "-Task")
 
-    def _resolve_start_soon(self, start_soon: bool | Sentinel) -> bool:
+    def _resolve_start_soon(self, start_soon: bool | None | Sentinel) -> bool:
```

Ruff (0.15.5)

[warning] Boolean-typed positional argument in function definition

(FBT001)